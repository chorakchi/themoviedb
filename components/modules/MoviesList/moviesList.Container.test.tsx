// @flow

import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import { MoviesList } from "./moviesList.Container";
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";
import thunk from "redux-thunk";
import { mount } from "enzyme";
import { createStore } from "redux";
import { INITIAL_STATE as reducer } from "./moviesList.reducer";

describe("MoviesList unit test", () => {
  it("should render correctly", () => {
    const mockStore = configureMockStore([thunk]);
    const store = mockStore({
      moviesList: reducer,
    });
    const subject = renderer
      .create(
        <Provider store={store}>
          <MoviesList />
        </Provider>
      )
      .toJSON();
    expect(subject).toMatchSnapshot();
  });
});
