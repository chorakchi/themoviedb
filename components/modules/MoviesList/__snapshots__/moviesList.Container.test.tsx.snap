// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesList unit test should render correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "paddingBottom": 20,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 20,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 34,
          },
        ]
      }
      variant="h4"
    >
      Movies
    </Text>
  </View>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={0.5}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</RCTSafeAreaView>
`;
